### ISSUE TITLE
Some suggestions for improving the experience on mobile

### ISSUE THREAD

**Issue by shiki-tm** [killergerbah/asbplayer#553]
I would like to suggest also adding the "open app" button or something similar to make it easy to get to the local video player. It's very hard to know that on mobile. 

And I know you're limited quite a bit by the browser apps but I don't think the supported codecs on mobile browsers is posted anywhere so I think if you could clarify somewhere maybe on the GitHub page or better in the local video app site, which audio and video codecs work on kiwi for example, it would help make things clearer for local files.

Also when playing a local video file with subtitles if you go into Fullscreen there's no overlay UI to help you mine anki cards, and the only way to mine is with the side panel but on a mobile screen this of course is in the way and kind of janky. Would it be possible to just have the same overlay for online videos show up when in the local video player website ? 
![Screenshot_20241025_113627_Kiwi Browser](https://github.com/user-attachments/assets/421752d4-696f-4d0e-a489-3a8b21fc36fe)

When I connected my Bluetooth keyboard I could press play and pause with the space key but no shortcuts seem to work. Is there any hope to see keyboard shortcuts working on mobile in the future? 

Thank you for the continued work on this project!

**Comment by killergerbah** [comment]
@shiki-tm Regarding keyboard shortcuts, those need to be bound through the browser at kiwi://extensions/shortcuts
This is also the page you should be redirected to when you try to edit the keyboard shortcuts through the app/extension settings. Have you bound those already?

**Comment by shiki-tm** [comment]
I was using the default shortcuts from that page like Ctrl shift U and Ctrl shift X

**Comment by killergerbah** [comment]
I see, can you confirm that those shortcuts are actually set?
In my case, when I open the extension shortcuts on Kiwi, all of the keyboard shortcuts are unbound by default.

**Comment by shiki-tm** [comment]
Yeah looks like they're set, I never had to do anything on this page.
![Screenshot_20241025_221749_Kiwi Browser](https://github.com/user-attachments/assets/a1c36756-9620-4547-b996-e85e4c9f470f)

**Comment by killergerbah** [comment]
Got it, thanks for checking!
It's not obvious to me why keyboard shortcuts are not working on mobile as the code for the Kiwi build is exactly the same as desktop. So it may take a while for me to find the fix.

**Comment by killergerbah** [comment]
Hi @shiki-tm , I believe I've addressed this issue:

> Also when playing a local video file with subtitles if you go into Fullscreen there's no overlay UI to help you mine anki cards, and the only way to mine is with the side panel but on a mobile screen this of course is in the way and kind of janky. Would it be possible to just have the same overlay for online videos show up when in the local video player website ?

Would be great to get your feedback.

**Comment by shiki-tm** [comment]
I just lost my comment. Gonna write this again... 
Sorry for the late reply ive been so busy but I did try this out like a day or two ago. Basically I think this is great, it's going in the right direction and I really appreciate you giving this time. I just have to point a serious bug and maybe a suggestion or two. 

![Screenshot_20241221_111349_Kiwi Browser](https://github.com/user-attachments/assets/0126da03-92c0-4dce-9402-6b9788366e6a)
So the problem is with the button on the far right there. I think it's for checking the mining history? Just when there's any subtitle or video files, if tapped on it will crash the browser lmao. I think it's kind of major and could just turn people off from the idea of ever trying to mine on their phone or tablet with asbplayer because they could just assume this doesn't work at all for mobile. 

https://github.com/user-attachments/assets/3fbccf98-f3df-4a9c-aecd-3cee8f203665

And then the screen recording shows how you have to scroll down to see the overlay. At first I couldn't actually see it and was very confused. I think going into fullscreen would make this fine but the player's fullscreen button doesn't actually go fullscreen all the way. I'm not sure if this can be even be addressed if it's a browser limitation but I did find that on kiwi browser and edge canary, you can make PWA. I think that's what they're called. "Add to home screen". With this, the local app is always in Fullscreen and just seems to be a bit smoother overall experience. So if the Fullscreen thing can't really be addressed in the online page version then maybe suggesting to people who want to use this on their phone or tablet to install the local player as a PWA is probably a good idea. 

Also being able to switch between audio tracks would be great but idk if there's a browser limitation to this or not. Of course I'd love to see almost all the other buttons on there like switching subs position and multiple subs and all that but I also understand you don't want to bloat this mobile screen too much. 

Another thing that's not so easy to realize is about loading subtitles. It's just less noticeable on a small screen, knowing what to press to add subs, especially if you don't load the the sub file together with the video file. Idk if this is possible but maybe when on mobile, if there's no subs loaded you can have something over the open file (or another) button, that says "add/load subtitle". Hm, this suggestion is maybe not the best but I think you understand my point. I'm going to test this much more next week when I have a bit more free time and I'm sure I'll have more suggestions for you then. Really thank you so much!

**Comment by killergerbah** [comment]
@shiki-tm Hey thanks for the really good feedback. These are lot of things that I completely missed so if you have anything else feel free to keep it coming.

> So the problem is with the button on the far right there. I think it's for checking the mining history? Just when there's any subtitle or video files, if tapped on it will crash the browser lmao. I think it's kind of major and could just turn people off from the idea of ever trying to mine on their phone or tablet with asbplayer because they could just assume this doesn't work at all for mobile.

Thanks, this is a pretty bad issue and fix is coming.

> And then the screen recording shows how you have to scroll down to see the overlay. At first I couldn't actually see it and was very confused. I think going into fullscreen would make this fine but the player's fullscreen button doesn't actually go fullscreen all the way. I'm not sure if this can be even be addressed if it's a browser limitation but I did find that on kiwi browser and edge canary, you can make PWA. I think that's what they're called. "Add to home screen". With this, the local app is always in Fullscreen and just seems to be a bit smoother overall experience. So if the Fullscreen thing can't really be addressed in the online page version then maybe suggesting to people who want to use this on their phone or tablet to install the local player as a PWA is probably a good idea.

Agreed, pushing to PWA makes sense. It turns out asbplayer is already installable as a PWA so all that's left is to implement a dialog to get people to install it.

> Also being able to switch between audio tracks would be great but idk if there's a browser limitation to this or not. Of course I'd love to see almost all the other buttons on there like switching subs position and multiple subs and all that but I also understand you don't want to bloat this mobile screen too much.
> (But this could also be helped a lot if the keyboard shortcuts work on mobile)

Yes audio track selection does require a browser flag to be flipped but it does make sense to add these to the mobile controls.

> Another thing that's not so easy to realize is about loading subtitles. It's just less noticeable on a small screen, knowing what to press to add subs, especially if you don't load the the sub file together with the video file. Idk if this is possible but maybe when on mobile, if there's no subs loaded you can have something over the open file (or another) button, that says "add/load subtitle". Hm, this suggestion is maybe not the best but I think you understand my point. I'm going to test this much more next week when I have a bit more free time and I'm sure I'll have more suggestions for you then. Really thank you so much!

Yes very good point. For now I've added a button to the top-left of the video player.

**Comment by killergerbah** [comment]
It looks like some more work is needed for asbplayer to work as an offline PWA. I'll work on that later.

**Comment by shiki-tm** [comment]
Some other things I noticed. Unfortunately I haven't been able to test this much still but I'm trying to find that time this weekend. 

![Screenshot_20250102_093124_Kiwi Browser](https://github.com/user-attachments/assets/51b12432-9008-4288-80a0-5461b85fccb5)
One thing that I don't know if it's a good idea to change is the fact that when you load subs, the position of the overlay is like right in the way of subs when they're at the bottom. They can be adjusted but just thought for beginners this would be troublesome and maybe changing the position of the overlay back to the top or automatically moving subs above it, could be a nice change. 


![Screenshot_20250103_001309_Kiwi Browser](https://github.com/user-attachments/assets/384582ad-f032-4a91-b3d7-f83415edf58d)
One thing you can't notice in my other screenshots is that notification/status bar. After taking a full screenshot this is how my screen really looks. It's not fullscreen all the way... Which is a minor thing I know but if it was really fullscreen like other pwa, it would be much more immersive in my opinion. 

Thank you

**Comment by killergerbah** [comment]
@shiki-tm Thanks again for your very on-point suggestions and observations.
I believe I've addressed both of your latest points:

> One thing that I don't know if it's a good idea to change is the fact that when you load subs, the position of the overlay is like right in the way of subs when they're at the bottom, at default. 

The subtitles will move out of the way now when the overlay appears.

> One thing you can't notice in my other screenshots is that notification/status bar.

The "real" fullscreen button is now available again on mobile in the top-right.

Would love to get your thoughts again whenever you have a chance.
